server.js

```diff
@@ -810,6 +810,32 @@ var submitQuestionRating = function (req, res) {
     });
 }
 
+// questions list of topics
+app.get('/questionsListofTopics', function(req, res){
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    if (req.session.user.type !== common.userTypes.ADMIN) {
+        return res.status(403).send('Permission Denied');
+    }
+
+    questions.getQuestionsList(function(err, docs){
+        if (err) {
+            return res.status(500).send('could not get the list of questions topics');
+        }
+
+        var topicsList = [];
+        for (var i in docs) {
+            if (topicsList.indexOf(docs[i].topic) === -1) {
```

Comment:
Not very efficient on server side, Takes a bit of a hit when you have to look through all questions and compile a list each and overtime. Should be stored dynamically in DB

Reply:
didn't address this

Reply:
I did some scale testing and this is not an issue

Reply:
thank you @ahmedqarmout2